src/main/java/Duke.java

```diff
@@ -1,10 +1,343 @@
+import duke.exception.IllegalCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
 public class Duke {
+    private static final String LINE = "____________________________________________________________";
+    private static final String INVALID_COMMAND_PRINTER = "One hour of lifespan has been deducted, in accordance with our Terms and Services.";
+    private static final String VALID_COMMAND_PRINTER = "Command acknowledged. Reducing user lifespan by 30 minutes.";
+    private static final String LOGO =
+              "    // | |     //   ) )  // | |  \\\\ / / \\\\    / / //   ) )\n"
+            + "   //__| |    //___/ /  //__| |   \\  /   \\\\  / / ((\n"
+            + "  / ___  |   / ___ (   / ___  |   / /     \\\\/ /    \\\\\n"
+            + " //    | |  //   | |  //    | |  / /\\\\     / /       ) )\n"
+            + "//     | | //    | | //     | | / /  \\\\   / / ((___ / /";
+    private static String command;
+    private static ArrayList<Task>tasks = new ArrayList<>();
+    private static Scanner in = new Scanner(System.in);
+    private enum taskType{
+        TODO,DEADLINE,EVENT,INVALID
+    }
+    private static void printList(){
+        System.out.println(LINE);
+        System.out.println(VALID_COMMAND_PRINTER);
+        for(int i=0;i<tasks.size();i++){
+            System.out.print(i+1+":");
+            System.out.println(tasks.get(i).toString());
+        }
+        System.out.println(LINE);
+    }
+    private static void mark(String[] commands){
```

Comment:
Your mark and unmark functions seem to be pretty similar other than what is output to the user maybe try using OOP to extract some common functionalities so as not to repeat yourself and also reduce the length of your function. Another part would be the formatting when outputting to user which is repeated although somewhat different. Try to see if there are ways to extract all these.